docs/docs/concepts/key-parameters.md

```diff
@@ -0,0 +1,85 @@
+
+To resolve this issue it is important to understand the difference between key parameters and immutable identifiers in Microsoft 365. The key parameters are the parameters that are used to identify a resource. The immutable identifiers are the unique identifiers that are created during the creation of the resource. These identifiers are not configurable and are not exposed to the user. The key parameters are used to identify the resource and the immutable identifiers are used to identify the resource within the platform.
+
+Changing the key parameters of those resource with duplicate keys is not an option as we would create a different configuration. In most cases the root cause is the fact, that we can have objects in Azure and Microsoft 365 with the same display name, but different immutable identifiers.
+
+We can differentiate between two types of resources:
+
+1. Those which have a unique immutable identifier and can be identified by the key parameters. Like the AADUser resource, which has a unique immutable identifier and can be identified by the UserPrincipalName.
+2. Those which have a unique immutable identifier, but can not be identified by the key parameters. Like the AADApplication resource, which has a unique immutable identifier, but can not be uniquely identified by the DisplayName.
+
+For 1. we can change the key parameters and the configuration will compile. We can rely on the immutable identifier to identify the resource. For 2. we can not rely on the immutable identifier to identify the resource. This is a big challenge for DSC, as we need to have a unique identifier prior to the creation of the objects in the tenant.
+
+## The current implementation of key parameters in Microsoft365DSC
+
+With the April 2023 release of Microsoft365DSC we have many breaking changes that help to better identify the resources in the configuration. There were several changes to key parameters. This implementation  a new way of handling key parameters. This new implementation is based on the following principles:
+
+1. The key parameters are the parameters that are used to identify a resource.
+2. The immutable identifiers are the unique identifiers that are created during the creation of the resource. Fom some resources, these are now exposed through M365DSC. This offers two use cases:
+   1. During the initial creation of the resource the display name would be used to identify the resource. This is the same as the current implementation.
+   2. For updates of the resource, the immutable identifier would be used to identify the resource. This is a new implementation.
+
+The new implementation is kind of a heuristic approach to enable exports and imports of configurations. Besides that it is a good practice to better identify the resources in the configuration.
```

Comment:
Maybe we can add a "More information" section with a link to this article:
https://learn.microsoft.com/en-us/powershell/dsc/resources/authoringresourcemof?view=dsc-1.1#the-contents-of-the-mof-file

> The type qualifier, [Key], on a property indicates that this property will uniquely identify the resource instance. At least one [Key] property is required.

Reply:
done